independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/mapping/RequestMapper.java

```diff
@@ -138,10 +156,18 @@ private RequestMatch<T> mapFromPathMatcher(String path, PathMatcher.PathMatch<Ar
                         remaining = path.substring(matchPos);
                     }
                 }
-                return new RequestMatch(potentialMatch.template, potentialMatch.value, params, remaining);
+                matches.add(new RequestMatch(potentialMatch.template, potentialMatch.value, params, remaining));
             }
         }
-        return null;
+
+        matches.sort(Comparator.comparingInt(new ToIntFunction<RequestMatch<T>>() {
```

Comment:
Why not saving the comparator into a static final field?

In theory the sort could be avoided since most of the caller just get the first element (List.get(0)), which means that the list could be iterated (or during adding matches too, without iterating it again) and retrieved the top element to return.

Reply:
No sort needed. I noticed that the list of `RequestPath`s is already sorted by length descending. 
So we already get our `matches` list in ordered by remaining length ascending.